test-d/required-deep.ts

```diff
@@ -0,0 +1,72 @@
+import {expectTypeOf} from 'expect-type';
+import {RequiredDeep} from '../index';
+
+interface Foo {
+	baz?: string | undefined;
+	bar?: {
+		function?: ((...args: any[]) => void) | undefined;
+		object?: {key?: 'value'} | undefined;
+		string?: string | undefined;
+		number?: number | undefined;
+		boolean?: false | undefined;
+		date?: Date | undefined;
+		regexp?: RegExp | undefined;
+		symbol?: Symbol | undefined;
+		null?: null | undefined;
+		undefined?: undefined;
+		map?: Map<string | undefined, string | undefined>;
+		set?: Set<string | undefined>;
+		array?: (string | undefined)[];
+		tuple?: ['foo' | undefined] | undefined;
+		readonlyMap?: ReadonlyMap<string | undefined, string | undefined>;
+		readonlySet?: ReadonlySet<string | undefined>;
+		readonlyArray?: readonly (string | undefined)[];
+		readonlyTuple?: readonly ['foo' | undefined] | undefined;
+	};
+}
+
+interface FooRequired {
+	baz: string;
+	bar: {
+		function: (...args: any[]) => void;
```

Comment:
```suggestion
		function: (...arguments: any[]) => void;
```

Reply:
The term `arguments` is not allowed when TypeScript is in strict mode anymore, if it ever way.